src/components/time-picker/time-picker.tsx

```diff
@@ -666,6 +674,32 @@ export class TimePicker {
     }
   };
 
+  private setMeridiemOrder(newLocale?: string): void {
+    if (this.isDefaultLocaleRTL(newLocale)) {
+      this.meridiemOrder = "-1";
+    } else {
+      this.meridiemOrder = this.getMeridiemOrder() === 0 ? "-1" : "0";
+    }
+  }
+
+  private getMeridiemOrder(): number {
+    if (this.formatParts) {
+      const index = this.formatParts.findIndex((parts: { type: string; value: string }) => {
+        return parts.value === this.localizedMeridiem;
+      });
+      return index;
+    }
+    return 0;
+  }
+
+  private isDefaultLocaleRTL(newLocale?: string): boolean {
```

Comment:
cc @jcfranco 

I don't think we should be having `locale` influence `dir` within components. Only the `dir` attribute should influence `direction` styling. `dir` and `lang` are two separate attributes in HTML that do not influence each other. Having one modify the other can become complex and inconsistent with how native HTML works. 

I think it's fine if we want to say that an element inside a component should always be styled `direction: ltr;`, but having a specific locale specify the direction styling opens up a can of worms of doing different things for different languages and styling becomes unpredictable.

I would push for not having `lang` influence `direction` styling.

Reply:
Fair points. FWIW, `direction` and `lang` were being set together to avoid incorrect display/behavior when one was set, but not the other. We confirmed w/ the translation team that we can assume users will set both when needed, so I think we can simplify this approach.

Reply:
Even with the assumption of setting `dir` for `lang` other than default, the current complexity is allowing page direction  for default `lang` and determining default `lang`. If we can able to ignore the direction irrespective of lang that would cut down the complexity.